### PR TITLE
SCANNPM-2 polish logging and cleanup invalid cache

### PR DESCRIPTION
**Pull request by lucas-paulger-sonarsource**

### CODE REVIEW

File: src/properties.ts
Comment (lucas-paulger-sonarsource) [inline src/properties.ts:1]: these changes were to reduce complexity, the main change is abstracting these functions 
```
populatePackageParams(packageJsonParams, pkg);
populateCoverageParams(packageJsonParams, pkg, projectBaseDir, sonarBaseExclusions);
populateTestExecutionParams(packageJsonParams, pkg, projectBaseDir);
```

File: src/file.ts [hunk 1]
```diff
@@ -31,20 +31,29 @@ import { CacheFileData, ScannerProperties, ScannerProperty } from './types';
...
   } else {
-    log(LogLevel.INFO, `No Cache found for ${filePath}`);
+    log(LogLevel.INFO, `No Cache found for ${alias}`);
     return null;
   }
 }
 
-export async function extractArchive(fromPath: string, toPath: string) {
-  log(LogLevel.INFO, `Extracting ${fromPath} to ${toPath}`);
+export async function extractArchive(fromPath: string, toPath: string, canonicalBasePath: string) {
```
Comment (vdiez) [inline src/file.ts:55]:
```suggestion
export async function extractArchive(fromPath: string, toPath: string) {
```

not needed any longer, right?

File: src/java.ts [hunk 2]
```diff
@@ -111,33 +113,38 @@ export async function fetchJRE(properties: ScannerProperties): Promise<string> {
...
+  log(LogLevel.INFO, `Downloaded ${JRE_ALIAS} to ${archivePath}`);
+
   try {
     await validateChecksum(archivePath, jreMetaData.sha256);
   } catch (error) {
     await fsExtra.remove(archivePath);
     throw error;
   }
-  await extractArchive(archivePath, jreDirPath);
+  await extractArchive(archivePath, jreDirPath, getParentCacheDirectory(properties));
```
Comment (vdiez) [inline src/java.ts:147]:
```suggestion
  await extractArchive(archivePath, jreDirPath);
```

File: src/scanner-cli.ts [hunk 3]
```diff
@@ -27,7 +27,7 @@ import {
   SCANNER_CLI_MIRROR,
   SCANNER_CLI_VERSION,
 } from './constants';
-import { extractArchive } from './file';
+import { extractArchive, getParentCacheDirectory } from './file';
```
Comment (vdiez) [inline src/scanner-cli.ts:30]:
```suggestion
import { extractArchive } from './file';
```

File: src/scanner-cli.ts [hunk 4]
```diff
@@ -100,7 +100,7 @@ export async function downloadScannerCli(properties: ScannerProperties): Promise
   await download(scannerCliUrl.href, archivePath, overrides);
 
   log(LogLevel.INFO, `Extracting SonarScanner CLI archive`);
-  await extractArchive(archivePath, installDir);
+  await extractArchive(archivePath, installDir, getParentCacheDirectory(properties));
```
Comment (vdiez) [inline src/scanner-cli.ts:103]:
```suggestion
  await extractArchive(archivePath, installDir);
```

File: test/unit/file.test.ts [hunk 5]
```diff
@@ -69,7 +71,7 @@ describe('file', () => {
         const archivePath = 'path/to/archive.zip';
         const extractPath = 'path/to/extract';
 
-        await extractArchive(archivePath, extractPath);
+        await extractArchive(archivePath, extractPath, 'canonicalBasePath');
```
Comment (vdiez) [inline test/unit/file.test.ts:74]:
```suggestion
        await extractArchive(archivePath, extractPath);
```

File: test/unit/file.test.ts [hunk 6]
```diff
@@ -121,7 +123,7 @@ describe('file', () => {
           }
         });
 
-        await extractArchive(mockFilePath, mockDestDir);
+        await extractArchive(mockFilePath, mockDestDir, 'path/to/dest/');
```
Comment (vdiez) [inline test/unit/file.test.ts:126]:
```suggestion
        await extractArchive(mockFilePath, mockDestDir);
```

**Review by vdiez — CHANGES_REQUESTED** [review]
just some changes related to the fact that `canonicalBasePath` is not used any more

File: test/unit/scanner-engine.test.ts [hunk 7]
```diff
@@ -22,10 +22,10 @@ import axios from 'axios';
 import MockAdapter from 'axios-mock-adapter';
 import { ChildProcess, spawn } from 'child_process';
 import fs from 'fs';
```
Comment (vdiez) [inline test/unit/scanner-engine.test.ts:24]: sorry, I missed that before. Let's make sure we only use fs-extra everywhere instead of native fs.

**Review by vdiez — APPROVED** [review]
just a minor comment, to mutualize usage of fs-extra everywhere. otherwise, LGTM!!

File: test/unit/scanner-engine.test.ts (hunk 7 above)
Comment (lucas-paulger-sonarsource) [reply test/unit/scanner-engine.test.ts]: done

**Pull request merged**